modules/core/04-channel/keeper/upgrade.go

```diff
@@ -377,3 +376,73 @@ func (k Keeper) constructProposedUpgrade(ctx sdk.Context, portID, channelID stri
 		LatestSequenceSend: nextSequenceSend - 1,
 	}, nil
 }
+
+// abortUpgrade will restore the channel state and flush status to their pre-upgrade state so that upgrade is aborted.
+// any unnecessary state is deleted. An error receipt is written, and the OnChanUpgradeRestore callback is called.
+func (k Keeper) abortUpgrade(ctx sdk.Context, portID, channelID string, err error) error {
+	if err == nil {
+		return errorsmod.Wrap(types.ErrInvalidUpgradeError, "cannot abort upgrade handshake with nil error")
+	}
+
+	upgrade, found := k.GetUpgrade(ctx, portID, channelID)
+	if !found {
+		return errorsmod.Wrapf(types.ErrUpgradeNotFound, "port ID (%s) channel ID (%s)", portID, channelID)
+	}
+
+	if err := k.restoreChannel(ctx, portID, channelID); err != nil {
```

Comment:
should we wrap this err with some sort of error indicating it came from restoring the channel? since the error returned from `RestoreChannel` is just if the channel was not found.

Reply:
yep good call